worlds/ladx/__init__.py

```diff
@@ -0,0 +1,421 @@
+        for r in self.multiworld.get_regions(self.player):
+            for loc in r.locations:
+                if isinstance(loc, LinksAwakeningLocation):
+                    assert(loc.item)
+                    # If we're a links awakening item, just use the item
+                    if isinstance(loc.item, LinksAwakeningItem):
+                        loc.ladxr_item.item = loc.item.item_data.ladxr_id
+
+                    # TODO: if the item name contains "sword", use a sword icon, etc
+                    # Otherwise, use a cute letter as the icon
+                    else:
+                        loc.ladxr_item.item = self.guess_icon_for_other_world(loc.item.name)
+                        loc.ladxr_item.custom_item_name = loc.item.name
+
+                    if loc.item:
+                        loc.ladxr_item.item_owner = loc.item.player
+                    else:
+                        loc.ladxr_item.item_owner = self.player
+
+                    # Kind of kludge, make it possible for the location to differentiate between local and remote items
+                    loc.ladxr_item.location_owner = self.player
+
+        rom_path = "Legend of Zelda, The - Link's Awakening DX (USA, Europe) (SGB Enhanced).gbc"
+        out_name = f"AP-{self.multiworld.seed_name}-P{self.player}-{self.multiworld.player_name[self.player]}.gbc"
+        out_file = os.path.join(output_directory, out_name)
+
+        rompath = os.path.join(output_directory, f"{self.multiworld.get_out_file_name_base(self.player)}.gbc")
+
+
+
+        parser = get_parser()
+        args = parser.parse_args([rom_path, "-o", out_name, "--dump"])
+
+        name_for_rom = self.multiworld.player_name[self.player]
+
+        all_names = [self.multiworld.player_name[i + 1] for i in range(len(self.multiworld.player_name))]
+        rom = generator.generateRom(
+            args,
+            self.laxdr_options,
+            self.player_options,
+            bytes.fromhex(self.multiworld.seed_name),
+            self.ladxr_logic,
+            rnd=self.multiworld.per_slot_randoms[self.player],
+            player_name=name_for_rom,
+            player_names=all_names,
+            player_id = self.player)
+      
+        handle = open(rompath, "wb")
+        rom.save(handle, name="LADXR")
+        handle.close()
+        patch = LADXDeltaPatch(os.path.splitext(rompath)[0]+LADXDeltaPatch.patch_file_ending, player=self.player,
+                                player_name=self.multiworld.player_name[self.player], patched_path=rompath)
+        patch.write()
+        if not DEVELOPER_MODE:
+            os.unlink(rompath)
+
+    def generate_multi_key(self):
+        return bytes.fromhex(self.multiworld.seed_name) + self.player.to_bytes(2, 'big')
+
+    def modify_multidata(self, multidata: dict):
+        multi_key = binascii.hexlify(self.generate_multi_key()).decode()
+        multidata["connect_names"][multi_key] = multidata["connect_names"][self.multiworld.player_name[self.player]]

```

Comment:
I see you encode a connect name, but also allow the client user entry of the player name?
I think that makes this the first game to do that, normally if a connect hash is encoded, that's the only way to connect with that game's client.